specs/node_agent_flows_and_namespaces.adoc

```diff
@@ -0,0 +1,623 @@
+    # parameters that need to be supplied for using that method. This being in
+    # the global tendrl namespace, defines the expected data types. These
+    # values would be used by the validations to ensure that wherever the
+    # `installation_sources` section is used, outside this global namespace, it
+    # conforms to the definitions provided.
+    #
+    # Each installation_type contains a list of mandatory parameters. Along
+    # with these, variables could be populated on the fly. These need to be
+    # supplied as `parameters` from the appropriate `installation` sections.
+    installation_sources:
+      # installation_type
+      yum_repository:
+        # Configure the yum repository. Doesn't actually install anything.
+        repository_path: String
+        gpgcheck: Boolean
+        packages: Hash
+      yum:
+        # Install packages using yum. Assume that the repositories have already
+        # been configured.
+        # The value supplied needs to be {name:version,name:version}
+        packages: Hash
+      git:
+        repository_path: String
+        # tag, branch or commit id
+        reference: String
+      pip_package:
+        package_name: String
+        # "latest" is a supported version, which will automatically use the
+        # latest available.
+        package_version: String
+      pip_requirements:
+        requirements_file: String
+      git+pip_requirements: Hash
+        # It is possible to chain multiple installation methods together. In
+        # which case, they'll be invoked in the order specified and the
+        # parameters would be required based on their individual definitions.
+      yum_repository+yum: Hash
+    platforms:
+      # platform_type supported by tendrl. One of these platforms and their
+      # corresponding versions must be reused in any of the definition files.
+      centos:
+        # platform_version
+        # Supported versions, in ascending order of release. This list, being
+        # ordered, enables the comparators used later in the file to function.
+        - 6.x
+        - 7.x
+      ubuntu:
+        - 15.04.x
+        - 15.10.x
+        - 16.04.x
+    invocations:
+      service:
+        # systemd service
+        # implementation module: tendrl.global.Invocations.Service
+        parameters:
+          name: String
+          start_action: String
+          stop_action: String
+      cli:
+        # Generic CLI command
+        # implementation module: tendrl.global.Invocations.Cli
+        parameters:
+          # Array of objects
+          - command: String
+            # Provide either a list of success or failure exit codes. Anything
+            # else would indicate the other condition.
+            success_exit_code: Array
+      module:
+        # A specific module in the code
+        parameters:
+          - module: Module
+            parameters: Hash
+    storage_system_types:
+      ceph: namespace.tendrl.integrations.ceph
```

Comment:
we call it "namespace.tendrl.ceph_integration", i suppose we have to change it to "namespace.tendrl.integrations.ceph"

same for gluster